src/yz_exchange_fsm.erl

```diff
@@ -122,27 +122,48 @@ prepare_exchange(start_exchange, S) ->
 prepare_exchange(timeout, S) ->
     do_timeout(S).
 
-update_trees(start_exchange, S=#state{yz_tree=YZTree,
-                                      kv_tree=KVTree,
+update_trees(start_exchange, S=#state{kv_tree=KVTree,
+                                      yz_tree = YZTree,
                                       index=Index,
                                       index_n=IndexN}) ->
-
-    update_request(yz_index_hashtree, YZTree, Index, IndexN),
-    update_request(riak_kv_index_hashtree, KVTree, Index, IndexN),
+    Self = self(),
+    update_request(
+        riak_kv_index_hashtree, KVTree, Index, IndexN,
+        fun() ->
```

Comment:
- If we move the `Callback` from https://github.com/basho/riak_kv/pull/1361 to within the `spawn_link`, we can probably remove the `spawn` here. 
-  Then, as per @tburghart's [doc](https://github.com/basho/internal_wiki/pull/6/files), we can keep the main fun that the Callback calls, but move the drain function below (again w/o the spawn) into a separate function that we can call dialyze correctly... and call it via `F = fun FunctionName/Arity` (i.e. a named function).